pkg/routes/mfx/mfx.go

```diff
@@ -30,12 +30,16 @@ type mfxRoute struct {
 	route routes.Route
 }
 
-func NewRoute(n, m, s string, log logger.Logger, pub publish.Publisher) routes.Route {
+func NewRoute(n, m, s string, w int, log logger.Logger, pub messages.Publisher) routes.Route {
 	return &mfxRoute{
-		route: routes.NewRoute(n, m, s, log, pub),
+		route: routes.NewRoute(n, m, s, w, log, pub),
 	}
 }
 
+func (mr *mfxRoute) Workers() int {
```

Comment:
Why do we need these getters? Why not exporting `Route` and then getting them directly from there? This looks like a redundant code "a la Java"

Reply:
Because Route can have different implemnation

Reply:
So?

Reply:
Route can have different implementation but some fields are common.
so there is Route which is like default implementation
and in mfx.go there is another implementation which differs in Consume method, all other properties are same.
mfx.New and route.New just return Route interface so I don't have to think about implementation just use interface.
Since the mfx has most of the methods and properties same as default implementation it contains object of default implementation ( inheritance with composition ) and just passes underlying values in methods as its own